ProvisionSample/Program.cs

```diff
@@ -588,29 +594,65 @@ static void Main(string[] args)
             return client;
         }
 
+        static async Task<IEnumerable<string>> GetTenantIdsAsync(string commonToken)
+        {
+            using (var httpClient = new HttpClient())
+            {
+                httpClient.DefaultRequestHeaders.Add("Authorization", "Bearer " + commonToken);
+                var response = await httpClient.GetStringAsync("https://management.azure.com/tenants?api-version=2016-01-29");
```

Comment:
Definitely agree for production use cases.  This is only a sample app to give basic guidance to our customers and demonstrates how to get the AAD tokens / tenants.